src/net_processing.cpp

```diff
@@ -4144,6 +4300,23 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         m_txrequest.ReceivedResponse(pfrom.GetId(), txid);
         if (tx.HasWitness()) m_txrequest.ReceivedResponse(pfrom.GetId(), wtxid);
 
+        if (m_opts.sensitiverelayowntx && m_mempool.RemoveUnbroadcastTx(txid)) {
+            LogPrintLevel( /* Continued */
+                BCLog::SENSITIVE_RELAY,
+                BCLog::Level::Info,
+                "Received own transaction (txid=%s) from the network from peer=%d%s; "
```

Comment:
_moving discussion from https://github.com/bitcoin/bitcoin/pull/27509#issuecomment-1659008138_

> 4. Finally, on signet I never got an INV for one of my own txs back. so getmempoolinfo : unbroadcastcount only ever increased. I notice that the tests only use python P2PInterface()'s for peers and the observer_outbound peer and is manually forced to inv our txid... makes me wonder if real bitcoind peers have different enough relay behavior that it doesnt work as well in production. For example, we pretend our sensitive relay peers requested a tx when they didn't. Does bitcoind still relay TXs it didn't getdata for? Also I suppose given that some of my signet TXs weren't going out, there could be an issue with orphans and peers that are missing my parents. I may try writing tests with more network bitcoind nodes, or set it up locally but routed through Tor.

The witness was not serialized, so the recipients rejected the transaction :face_with_spiral_eyes: I fixed this and updated the test to catch cases where `bitcoind` would not accept or further relay the transaction to other peers. Now the test has one more `bitcoind`, which is used as a recipient.

Reply:
🕺  ah nice! I thought there might be something wrong with the transactions themselves, but when I restarted without sensitive relay everything broadcast fine. I didn't think of serialization being the issue! So, great catch.